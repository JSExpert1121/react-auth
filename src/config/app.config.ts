export const KAKAOMAP_API_KEY = '44a509cf4bf161e250c940454df5977c';
export const SUPPORTED_COUNTRIES = ['kr', 'jp', 'cn', 'hk', 'sg', 'tw', 'th', 'my', 'au', 'us', 'gb', 'de', 'fr', 'kz', 'ae', 'in'];
// export const COUNTRY_CODES = ['82', '81', '86', '852', '65', '886', '66', '60', '61', '1', '44', '49', '49', '33', '7', '971', '91'];
