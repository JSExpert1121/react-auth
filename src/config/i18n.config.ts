/**
 * @name i18n.config
 * @description config file for i18n. currently support ko and en
 * @version 1.0
 * @todo add locales
 * @author Darko
 */
const enMessages = require('translations/en.json');
const koMessages = require('translations/ko.json');

export const DEFAULT_LOCALE = 'en';

export const appLocales = [
    { code: 'en', name: 'English' },
    { code: 'ko', name: 'Korean' }
];

// auxilliary function to generate messages for each locales
export const formatMessages: (locale: string, messages: Record<string, string>) => Record<string, string> = (locale, messages) => {
    const defaultFormattedMessages =
        locale !== DEFAULT_LOCALE
            ? formatMessages(DEFAULT_LOCALE, enMessages)
            : {};
    const flattenFormattedMessages = (formattedMessages: Record<string, string>, key: string) => {
        const formattedMessage =
            !messages[key] && locale !== DEFAULT_LOCALE
                ? defaultFormattedMessages[key]
                : messages[key];
        return Object.assign(formattedMessages, { [key]: formattedMessage });
    };
    return Object.keys(messages).reduce(flattenFormattedMessages, {});
};

export const translationMessages = {
    en: formatMessages('en', enMessages),
    ko: formatMessages('ko', koMessages),
};
