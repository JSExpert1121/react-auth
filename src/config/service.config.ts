export const API_GATEWAY_ADDRESS = 'https://apig.t.moria.hitit.xyz:8765';
export const OAUTH_CLIENT_ID = 'web';
export const OAUTH_CLIENT_SECRET = 'hititweb42!';