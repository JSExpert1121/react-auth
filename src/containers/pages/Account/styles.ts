import { Theme, createStyles } from '@material-ui/core/styles';
import Colors from 'assets/colors';

export const styleFn = (theme: Theme) => createStyles({
    root: {
        width: '100%',
        padding: theme.spacing(3),
        maxWidth: 512,
        minWidth: 480,
    },
    form: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        width: '100%',
        textAlign: 'center'
    },
    logo: {
        padding: theme.spacing(2, 0)
    },
    title: {
        fontSize: '1.5rem',
        color: 'rgba(0, 0, 0, 0.8)',
        padding: theme.spacing(2)
    },
    text: {
        fontSize: '0.875rem',
        color: 'rgba(0, 0, 0, 0.7)',
        padding: theme.spacing(2)
    },
    success: {
        color: 'rgba(0, 0, 0, 0.9)',
        padding: theme.spacing(1)
    },
    error: {
        fontSize: '0.875rem',
        color: Colors.error,
        padding: theme.spacing(0, 0, 0, 2)
    },
    busy: {
        width: '100%',
    },
    submit: {
        margin: theme.spacing(2, 0),
        height: 'calc(100% - 12px)'
    },
    bottomMenu: {
        width: '100%',
        display: 'flex',
        padding: theme.spacing(2, 0)
    }
});