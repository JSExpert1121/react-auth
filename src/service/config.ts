import { oauthClient } from './base';

export const setLanguage = (lang: string) => {
    oauthClient.setLanguage(lang);
}
