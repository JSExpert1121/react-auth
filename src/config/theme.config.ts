import { createMuiTheme } from '@material-ui/core/styles';

const theme = createMuiTheme({
	palette: {
		primary: {
			main: '#3f51b5',
			dark: '#303f9f',
		},
		secondary: {
			light: '#0033EE',
			main: '#1100BB',
			contrastText: '#000000',
		},
		background: {
			default: '#FFF'
		}
	},
	typography: {
		fontFamily: "\"Noto Sans\", \"Noto Sans KR\", sans-serif"
	},
	overrides: {
		MuiAppBar: {
			colorPrimary: {
				color: '#111',
				backgroundColor: 'white'
			}
		},
		MuiContainer: {
			root: {
				padding: 0
			}
		},
		MuiBadge: {
			colorSecondary: {
				color: 'white'
			}
		},
		MuiExpansionPanel: {
			root: {
				margin: '8px 0',
				'&:before': {
					height: 0
				}
			}
		},
		MuiExpansionPanelSummary: {
			root: {
				padding: 0,
				'&:before': {
					height: 0
				}
			},
			content: {
				margin: 0,
				flexGrow: 0,
				'&.Mui-expanded': {
					margin: 0
				}
			}
		},
		MuiExpansionPanelDetails: {
			root: {
				padding: 8
			}
		}
	}
});

export default theme;
