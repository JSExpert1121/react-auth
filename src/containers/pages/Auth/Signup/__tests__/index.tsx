import React from 'react';
import { MemoryRouter, withRouter } from 'react-router-dom';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build';

import { createMount } from '@material-ui/core/test-utils';
import { ThemeProvider } from '@material-ui/styles';
import { IntlProvider, injectIntl } from "react-intl";
import { act } from 'react-dom/test-utils';

import DefSignupPage, { SignupPage } from '../index';
import SignupPage1 from '../Page1';
import SignupPage2 from '../Page2';

import { translationMessages } from 'config/i18n.config';
import theme from 'config/theme.config';

// store mock
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
import {
    GLOBAL_SET_TITLE,
} from 'store/constants';
jest.mock('service/member');

configure({ adapter: new Adapter() });
const InjectedSignupPage = withRouter(injectIntl(SignupPage));
const ConnectedSignupPage = withRouter(DefSignupPage);

const delay = ms => new Promise(resolve => setTimeout(resolve, ms));

describe('containers/pages/auth/signup', () => {

    const setTitle = jest.fn();
    const signup = jest.fn();
    const refresh = jest.fn();
    const store = mockStore({
        user: undefined,
        token: undefined
    });
    signup.mockResolvedValueOnce(undefined).mockRejectedValueOnce('Wrong credentials');

    // strange warning message ignore
    const mockConsoleMethod = (realConsoleMethod) => {
        const ignoredMessages = ['test was not wrapped in act(...)'];
        return (message, ...args) => {
            const containsMessage = ignoredMessages.some(ignoredMessage => message.includes(ignoredMessage))
            if (!containsMessage) {
                realConsoleMethod(message, ...args);
            }
        }
    }

    console.warn = jest.fn(mockConsoleMethod(console.warn));
    console.error = jest.fn(mockConsoleMethod(console.error));

    const mount = createMount();
    const wrapper = mount(
        <Provider store={store}>
            <MemoryRouter>
                <IntlProvider locale='en' messages={translationMessages.en}>
                    <ThemeProvider theme={theme}>
                        {<InjectedSignupPage setTitle={setTitle} signup={signup} refreshToken={refresh} />}
                    </ThemeProvider>
                </IntlProvider>
            </MemoryRouter>
        </Provider>
    );

    test('Snapshot test', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('UI components', () => {
        expect(wrapper.find(SignupPage).state().page).toBe(0);
        expect(wrapper.find(SignupPage1)).toHaveLength(1);
        expect(wrapper.find(SignupPage2)).toHaveLength(0);
        expect(wrapper.find(SignupPage1).props().goNext).toBeInstanceOf(Function);
    });

    test('componentDidMount', async done => {

        expect(setTitle).toHaveBeenCalledWith('Sign up');

        await delay(20);
        setImmediate(async () => {
            await wrapper.update();
            expect(wrapper.find(SignupPage).state().TAC).toHaveLength(5);
            done();
        });
    });

    test('downloadTAC', async done => {

        await wrapper.find(SignupPage).instance().downloadTAC('UNITTEST');

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find(SignupPage).state().TAC).toHaveLength(2);
            done();
        });
    });

    test('downloadTAC', async done => {

        await wrapper.find(SignupPage).instance().downloadTAC('UNITTEST1');

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find(SignupPage).state().TAC).toHaveLength(0);
            done();
        });
    });

    test('next page', done => {
        act(() => {
            wrapper.find(SignupPage).instance().handleNext('1', '2', '3', '3', '4', '5', true);
        });

        setImmediate(() => {
            wrapper.update();
            const Page = wrapper.find(SignupPage);
            expect(wrapper.find(SignupPage1)).toHaveLength(0);
            expect(wrapper.find(SignupPage2)).toHaveLength(1);
            expect(Page.state().page).toBe(1);
            done();
        });
    });

    test('back page', done => {
        act(() => {
            wrapper.find(SignupPage).instance().handleBack();
        });

        setImmediate(() => {
            wrapper.update();
            const Page = wrapper.find(SignupPage);
            expect(wrapper.find(SignupPage1)).toHaveLength(1);
            expect(wrapper.find(SignupPage2)).toHaveLength(0);
            expect(Page.state().page).toBe(0);
            done();
        });
    });

    test('signup', async done => {
        const result = await wrapper.find(SignupPage).instance().handleSignup([true]);

        setImmediate(() => {
            expect(signup).toHaveBeenCalledWith('1', '3', '2', '4', '5', [true]);
            expect(refresh).toHaveBeenCalledTimes(1);
            expect(result).toBe(true);
            done();
        });
    });

    test('signup failed', async done => {
        const result = await wrapper.find(SignupPage).instance().handleSignup([true]);

        setImmediate(() => {
            expect(signup).toHaveBeenCalledWith('1', '3', '2', '4', '5', [true]);
            expect(result).toBe(false);
            done();
        });
    });
});

describe('Connected Signup page', () => {
    const mount = createMount();

    const store = mockStore({
        user: undefined,
        token: undefined
    });

    const wrapper = mount(
        <Provider store={store}>
            <MemoryRouter>
                <IntlProvider locale='en' messages={translationMessages.en}>
                    <ThemeProvider theme={theme}>
                        {<ConnectedSignupPage />}
                    </ThemeProvider>
                </IntlProvider>
            </MemoryRouter>
        </Provider>
    );

    test('setTitle', done => {
        wrapper.update();
        expect(store.getActions()).toHaveLength(1);
        expect(store.getActions()).toContainEqual({
            type: GLOBAL_SET_TITLE,
            payload: 'Sign up'
        });
        done();
    });

    test('signup action', async done => {

        act(() => {
            wrapper.find(SignupPage).instance().handleNext('1', '2', '3', '3', '4', '5', true);
        });

        const result = await wrapper.find(SignupPage).instance().handleSignup([true]);
        setImmediate(() => {
            expect(result).toBe(true);
            done();
        });
    })
});