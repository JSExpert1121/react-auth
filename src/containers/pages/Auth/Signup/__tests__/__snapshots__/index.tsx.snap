// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/pages/auth/signup Snapshot test 1`] = `ReactWrapper {}`;
